apps/intellij/src/main/kotlin/dev/nx/console/toolWindow/NxProjectsTreeStructure.kt

```diff
@@ -0,0 +1,327 @@
+package dev.nx.console.toolWindow
+
+import com.intellij.execution.Executor
+import com.intellij.execution.RunManager
+import com.intellij.execution.RunnerAndConfigurationSettings
+import com.intellij.execution.executors.DefaultRunExecutor
+import com.intellij.execution.impl.RunDialog
+import com.intellij.icons.AllIcons
+import com.intellij.lang.javascript.JavaScriptBundle
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.DumbAware
+import com.intellij.ui.ClickListener
+import com.intellij.ui.PopupHandler
+import com.intellij.ui.tree.AsyncTreeModel
+import com.intellij.ui.tree.StructureTreeModel
+import com.intellij.ui.treeStructure.CachingSimpleNode
+import com.intellij.ui.treeStructure.SimpleNode
+import com.intellij.ui.treeStructure.SimpleTreeStructure
+import com.intellij.util.ui.tree.TreeUtil
+import dev.nx.console.NxIcons
+import dev.nx.console.models.NxProject
+import dev.nx.console.models.NxWorkspace
+import dev.nx.console.run.NxCommandConfiguration
+import dev.nx.console.run.NxCommandConfigurationType
+import dev.nx.console.run.NxRunSettings
+import java.awt.event.MouseEvent
+
+class NxProjectsTreeStructure(
+    val nxExecutor: NxExecutor,
+    val tree: NxProjectsTree,
+    val parentDisposable: Disposable,
+    private var nxWorkspace: NxWorkspace?
+) : SimpleTreeStructure() {
+
+    private val treeModel = StructureTreeModel(this, parentDisposable)
+    private var root: NxSimpleNode.Root = NxSimpleNode.Root(nxWorkspace)
+
+    init {
+        tree.model = AsyncTreeModel(treeModel, parentDisposable)
+        TreeUtil.installActions(tree)
+        installPopupActions()
+    }
+
+    override fun getRootElement(): Any = root
+
+    fun updateNxProjects(nxWorkspace: NxWorkspace?) {
+        this.nxWorkspace = nxWorkspace
+        root = NxSimpleNode.Root(nxWorkspace)
+        treeModel.invalidate()
+    }
+
+    sealed class NxSimpleNode(parent: SimpleNode?) : CachingSimpleNode(parent) {
+        class Root(val nxWorkspace: NxWorkspace?) : NxSimpleNode(null) {
+
+            init {
+                icon = NxIcons.Action
+            }
+
+            override fun buildChildren(): Array<SimpleNode> {
+                if (nxWorkspace == null) {
+                    return emptyArray()
+                }
+                val tasks: Array<SimpleNode> = arrayOf(Tasks(nxWorkspace, this))
+                val apps: Array<SimpleNode> = arrayOf(Apps(nxWorkspace, this))
+                val libs: Array<SimpleNode> = arrayOf(Libs(nxWorkspace, this))
+                return tasks + apps + libs
+            }
+
+            // TODO do not add a root node or get the name of workspace properly?
```

Comment:
I think adding a root node is a good idea, as it is the same as gradle toolwindow does it. It's an interesting question tho, how to get the name of a workspace. I think for now this is an okay solution. You could also use `project.nxBasepath` but it doesn't really matter.

Reply:
I was using the value npmScope from the json object but not sure if it's the intended value
We can use maybe project.name when nx monorepo is the root...
But let keep it simple